java-connectors/kafka-connect-query-language/src/main/java/io/lenses/kcql/Kcql.java

```diff
@@ -202,6 +203,27 @@ public Map<String, String> getProperties() {
     return properties;
   }
 
+  // TODO: Jira LC-203 improvements
+  public void validateKcqlProperties(String... allowedKeys) {
+
+    Set<String> unexpectedKeys =
+        properties.keySet().stream().filter(k -> !Arrays.stream(allowedKeys).collect(Collectors.toUnmodifiableSet())
+            .contains(k)).collect(Collectors.toUnmodifiableSet());
+    if (!unexpectedKeys.isEmpty()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unexpected properties found: `%s`. Please check the documentation to find the properties you really need.",
```

Comment:
This is a helpful behaviour for detecting misconfiguration.

We have been in the situation where connector users misspell a property.  It is better to fail fast so that the user can correct the configuration instead of leaving them frustrated not sure why the behaviour doesn't match what they've configured.

This also is the behaviour for the Scala connectors.

This links directly into the KCQL schemas we have planned (we have this already for the scala connectors) where each connector has a schema and any KCQL that doesn't fit in with what's expected in the schema will prompt an error.

Reply:
+1 on what @davidsloan on fail fast and avoid time on debugging. 

Connectors won't fail for a new configuration because it cannot know the required configurations for converters, SMTs. They are described by the owner of those plugins.